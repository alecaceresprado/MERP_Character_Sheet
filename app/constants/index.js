export * from './classes';
export * from './objects';
export * from './races';
