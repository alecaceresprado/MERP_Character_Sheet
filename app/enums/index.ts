export * from './races.enum';
export * from './objects.enum';
