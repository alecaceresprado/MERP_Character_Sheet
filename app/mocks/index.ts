export * from './characters.mock';
export * from './list.mock';
