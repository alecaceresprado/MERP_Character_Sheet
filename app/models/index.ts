export * from './player.model';
export * from './repo.model';
export * from './player-summary.model';
