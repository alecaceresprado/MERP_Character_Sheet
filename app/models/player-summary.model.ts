export interface PlayerSummary {
  id: string;
  description: string;
  avatarUrl: string;
  name: string;
}
