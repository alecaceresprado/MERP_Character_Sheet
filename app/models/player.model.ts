import { RacesEnum } from '../enums'

export interface Player {
    details: PlayerDetails;
    stats: PlayerStats;
    skills: PlayerSkills;
    inventory: PlayerInventory[];
}

export interface PlayerDetails {
    name: string,
    nickName: string,
    race: RacesEnum.NORSEMEN,
    height: string,
    weight: string,
    hairColor: string,
    eyesColor: string,
    physicalSpecial: string,
    class: string,
    classId: string,
    experience: 31751,
    level: 3,
}

export interface PlayerStats {
    strength: {
        num: number,
        normal: number,
        race: number,
        total: number,
    },
    agility: {
        num: number,
        normal: number,
        race: number,
        total: number,
    },
    stamina: {
        num: number,
        normal: number,
        race: number,
        total: number,
    },
    intellect: {
        num: number,
        normal: number,
        race: number,
        total: number,
    },
    intuition: {
        num: number,
        normal: number,
        race: number,
        total: number,
    },
    will: {
        num: number,
        normal: number,
        race: number,
        total: number,
    },
    beauty: {
        num: number,
        normal: number,
        race: number,
        total: number,
    },
}

export interface PlayerSkills {
    armor: {
        noArmor: {
            skillPoints: number,
        },
        leather: {
            skillPoints: number,
        },
        hardenedLeather: {
            skillPoints: number,
        },
        chainMail: {
            skillPoints: number,
        },
        plate: {
            skillPoints: number,
        },
    },
    weapons: {
        edge: {
            skillPoints: number,
        },
        blunt: {
            skillPoints: number,
        },
        twoHands: {
            skillPoints: number,
        },
        throw: {
            skillPoints: number,
        },
        projectile: {
            skillPoints: number,
        },
        dualWeapons: {
            skillPoints: number,
        },
        pole: {
            skillPoints: number,
        },
    },
    exterior: {
        climb: {
            skillPoints: number,
        },
        mount: {
            skillPoints: number,
        },
        swim: {
            skillPoints: number,
        },
        tracking: {
            skillPoints: number,
        },
    },
    survival: {
        ambush: {
            skillPoints: number,
        },
        hide: {
            skillPoints: number,
        },
        lockPick: {
            skillPoints: number,
        },
        trapDismantle: {
            skillPoints: number,
        },
    },
    magic: {
        readRunes: {
            skillPoints: number,
        },
        useObject: {
            skillPoints: number,
        },
        aimedSpell: {
            skillPoints: number,
        },
    },
    global: {
        sight: {
            skillPoints: number,
        },
        health: {
            skillPoints: number,
            pointValues: number[],
            sum: number,
        },
        mana: {
            skillPoints: number,
            pointValues: number[],
            sum: number,
        },
    },
    secondary: {
        blacksmith: {
            skillPoints: number,
        },
    },
}

export interface PlayerInventory {
    name: string;
    type: string;
    skill?: string[][];
    modifier?: number;
    description?: string;
    quantity?: number;
}
