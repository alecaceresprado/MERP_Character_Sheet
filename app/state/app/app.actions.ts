import { action, ActionType } from 'typesafe-actions';

import { PlayerSummary } from 'models';

export enum AppActionTypes {
  LOAD_PLAYERS_SUMMARY = '[APP] Load Players Summary',
  LOAD_PLAYERS_SUMMARY_SUCCESS = '[APP] Loading Players Summary With Sucess',
  LOAD_PLAYERS_SUMMARY_ERROR = '[APP] Error While Loading Players Summary'
}

export const AppActions = {
  loadPlayersSummary: () => action(AppActionTypes.LOAD_PLAYERS_SUMMARY),
  playersSummaryLoaded: (players: PlayerSummary[]) =>
    action(AppActionTypes.LOAD_PLAYERS_SUMMARY_SUCCESS, players),
  playersSummaryLoadingError: (error: object) =>
    action(AppActionTypes.LOAD_PLAYERS_SUMMARY_ERROR, error)
};

export type AppActionsTypes = ActionType<typeof AppActions>;
