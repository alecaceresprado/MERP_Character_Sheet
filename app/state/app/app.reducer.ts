import { PlayerSummary } from 'models';
import { AppActionsTypes, AppActionTypes } from './app.actions';

export interface AppState {
  readonly loading: boolean;
  readonly error?: object | boolean;
  readonly players: PlayerSummary[];
}

export const appInitialState: AppState = {
  loading: false,
  error: false,
  players: []
};

// Take this container's state (as a slice of root state), this container's actions and return new state
export function appReducer(
  state: AppState = appInitialState,
  action: AppActionsTypes
): AppState {
  switch (action.type) {
    case AppActionTypes.LOAD_PLAYERS_SUMMARY:
      return {
        ...state,
        loading: true,
        error: false,
        players: []
      };
    case AppActionTypes.LOAD_PLAYERS_SUMMARY_SUCCESS:
      return {
        ...state,
        loading: false,
        players: action.payload
      };
    case AppActionTypes.LOAD_PLAYERS_SUMMARY_ERROR:
      return {
        ...state,
        error: action.payload,
        loading: false
      };
    default:
      return state;
  }
}
