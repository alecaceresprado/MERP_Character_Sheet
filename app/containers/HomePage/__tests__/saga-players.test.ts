/**
 * Tests for HomePage sagas
 */

import { put, takeLatest } from 'redux-saga/effects';

import { AppActions, AppActionTypes } from 'state/app/app.actions';
import getPlayerData, { getPlayersSummary } from '../saga-players';
import { PlayerSummary } from 'models';

describe('getPlayersSummary Saga', () => {
  let getPlayersSummaryGenerator;

  // We have to test twice, once for a successful load and once for an unsuccessful one
  // so we do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    getPlayersSummaryGenerator = getPlayersSummary();

    const selectDescriptor = getPlayersSummaryGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should dispatch the playersSummaryLoaded action if it requests the data successfully', () => {
    const response = [
      {
        id: '1',
        description: 'Braided long beard with charms with runes',
        avatarUrl:
          'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcT3JvT2n7sHUoBw9QqyV-KlwYHfKphD4y0thI9k_iJsyeKmWwcZ&usqp=CAU',
        name: 'Lothar Picco'
      },
      {
        id: '2',
        description: 'Braided long beard with charms with runes',
        avatarUrl:
          'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcT3JvT2n7sHUoBw9QqyV-KlwYHfKphD4y0thI9k_iJsyeKmWwcZ&usqp=CAU',
        name: 'Lothar Picco'
      },
      {
        id: '3',
        description: 'Braided long beard with charms with runes',
        avatarUrl:
          'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcT3JvT2n7sHUoBw9QqyV-KlwYHfKphD4y0thI9k_iJsyeKmWwcZ&usqp=CAU',
        name: 'Lothar Picco'
      },
      {
        id: '4',
        description: 'Braided long beard with charms with runes',
        avatarUrl:
          'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcT3JvT2n7sHUoBw9QqyV-KlwYHfKphD4y0thI9k_iJsyeKmWwcZ&usqp=CAU',
        name: 'Lothar Picco'
      }
    ] as PlayerSummary[];
    const putDescriptor = getPlayersSummaryGenerator.next(response).value;
    expect(putDescriptor).toEqual(
      put(AppActions.playersSummaryLoaded(response))
    );
  });

  it('should call the playersSummaryLoadingError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getPlayersSummaryGenerator.throw(response).value;
    expect(putDescriptor).toEqual(
      put(AppActions.playersSummaryLoadingError(response))
    );
  });
});

describe('getPlayerData Saga', () => {
  const getPlayerDataSaga = getPlayerData();

  it('should start task to watch for LOAD_REPOS action', () => {
    const takeLatestDescriptor = getPlayerDataSaga.next().value;
    expect(takeLatestDescriptor).toEqual(
      takeLatest(AppActionTypes.LOAD_PLAYERS_SUMMARY, getPlayersSummary)
    );
  });
});
