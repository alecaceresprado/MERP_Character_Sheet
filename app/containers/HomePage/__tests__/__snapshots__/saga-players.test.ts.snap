// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPlayersSummary Saga should call the playersSummaryLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/mock",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getPlayersSummary Saga should dispatch the playersSummaryLoaded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/mock",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
